CRM/Contribute/BAO/Contribution.php

```diff
@@ -237,11 +236,11 @@ public static function add(&$params, $ids = []) {
    *
    * @param array $params
    *
-   * @return bool
+   * @return bool|int
    */
   public static function isUpdateToRecurringContribution($params) {
     if (!empty($params['contribution_recur_id']) && empty($params['id'])) {
-      return TRUE;
+      return (int) $params['contribution_recur_id'];
```

Comment:
I think we need to rename this function if it's not longer about getting a bool.

Reply:
@eileenmcnaughton Happy to rename, but can't think of a better name - I thought it sensible to return the recurID rather than have to run that logic again.